### PR TITLE
[ROCM] Enable more dtypes in common_method_invocations

### PR DESCRIPTION
**Pull request by arindamroy-eng**
The PR enables additional dtypes in common_method_invocations for ROCM.
This enables around 4k new tests for ROCM.

### CODE REVIEW

**Comment by arindamroy-eng** [comment]
@jithunnair-amd

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -2599,7 +2599,8 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            # BFloat16 support on CUDA requires CUDA 11 and SM53
            dtypesIfCUDA=floating_types_and(torch.float16, torch.complex64, torch.complex128,
                                            *[torch.bfloat16] if CUDA11OrLater else []),
-           dtypesIfROCM=floating_types_and(torch.half),
+           dtypesIfROCM=floating_types_and(torch.float16, torch.complex64, torch.complex128,
+                                           *[torch.bfloat16] if CUDA11OrLater else []),
```
Comment (jithunnair-amd) [inline torch/testing/_internal/common_methods_invocations.py:2603]: Does this condition `if CUDA11OrLater else []` make sense for ROCm?

Comment (arindamroy-eng) [reply torch/testing/_internal/common_methods_invocations.py]: Sorry for that. Removed the check and just added bfloat16.

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -280,7 +280,7 @@ def supported_dtypes(self, device_type):
         if device_type == 'cpu':
             return self.dtypesIfCPU
         if device_type == 'cuda':
-            return self.dtypesIfROCM if TEST_WITH_ROCM else self.dtypesIfCUDA
+            return self.dtypesIfCUDA
```
Comment (jithunnair-amd) [inline torch/testing/_internal/common_methods_invocations.py:283]: It looks like `dtypesIfROCM` is still set and used, so why does this code block not need to use it?

Comment (arindamroy-eng) [reply torch/testing/_internal/common_methods_invocations.py:283]: Hi @jithunnair-amd ,
This basically sets that the dtypes will be equal to for that of CUDA.
So there is always parity between ROCM and CUDA.

**Review by ngimel — APPROVED** [review]

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (walterddr) [reply torch/testing/_internal/common_methods_invocations.py:283]: Hmm. This doesn't seem consistent with what this PR does (e.g. some `dtypesIfROCM` were still changed

Comment (jithunnair-amd) [reply torch/testing/_internal/common_methods_invocations.py:283]: Yes, precisely, hence my comment above. Let me follow up offline with the PR author to get a bit more context.

**Pull request closed without merging**